packages/a11y-base/src/aria-hidden.js

```diff
@@ -123,7 +116,9 @@ const applyAttributeToOthers = (originalTarget, parentNode, markerName, controlA
       return;
     }
 
-    [...parent.children].forEach((node) => {
```

Comment:
This change is actually likely to cause problems as it will set `aria-hidden` on elements inside shadow DOM, while having no `aria-hidden` on the host. I'll try to find a better solution instead.

Reply:
UPD: added logic to keep `assignedSlot` - this resolves the issue with incorrect elements getting hidden.